test/stresses.jl

```diff
@@ -0,0 +1,33 @@
+# Hellmann-Feynman stress
+# via ForwardDiff & custom FFTW overloads on ForwardDiff.Dual
+using Test
+using DFTK
+using ForwardDiff
+import FiniteDiff
+
+@testset "ForwardDiff stresses on silicon" begin
+    function make_basis(a)
+        lattice = a / 2 * [[0 1 1.];
+                        [1 0 1.];
+                        [1 1 0.]]
+        Si = ElementPsp(:Si, psp=load_psp(:Si, functional="lda"))
+        atoms = [Si => [ones(3)/8, -ones(3)/8]]
+        model = model_DFT(lattice, atoms, [:lda_x, :lda_c_vwn]; symmetries=false)
+        kgrid = [4, 4, 4] # k-point grid (Regular Monkhorst-Pack grid)
+        Ecut = 7          # kinetic energy cutoff in Hartree
+        PlaneWaveBasis(model, Ecut; kgrid=kgrid, fft_size=[32, 32, 32])
+    end
+
+    function compute_energy(scfres_ref, a)
+        basis = make_basis(a)
+        energies, H = energy_hamiltonian(basis, scfres_ref.ψ, scfres_ref.occupation; ρ=scfres_ref.ρ)
```

Comment:
I don't think you can do this because the normalization of the rho depends on the volume of the unit cell. The orbitals are fine because they are l^2 normalized to 1, independently of the basis. You have to recompute rho in the new basis with the old orbitals.

Reply:
weird that the test doesn't pick that up. Can you try checking more carefully? Ie do the SCF to a very small precision and checking the result to more accuracy

Reply:
Also we should probably check it for a metal. I believe Hellman-Feynmann is also valid in this case but we should be careful.

Reply:
That was indeed a bug! Fixed it now by recomputing rho and added a consistency check against non-Hellmann-Feynman finite diff of the whole solve